### PR TITLE
Protectli vault ehl/vboot

### PR DESCRIPTION
**Pull request by Stojek139808**

### CODE REVIEW

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Comment by miczyg1** [comment]
@Stojek139808 layout looks good, do you any issues?

**Comment by Stojek139808** [comment]
@miczyg1 I was just wondering if we need to add an `unused` region. I had to do that last time when making a flash map, but there the flash had 32MB.

**Comment by miczyg1** [comment]
> @miczyg1 I was just wondering if we need to add an `unused` region. I had to do that last time when making a flash map, but there the flash had 32MB.

for 16MB we do not need any unused regions and gaps.

File: src/mainboard/protectli/vault_ehl/Kconfig [hunk 1]
```diff
@@ -28,4 +28,18 @@ config MAX_CPUS
 config CBFS_SIZE
 	default 0x900000
 
+config VBOOT
+	select GBB_FLAG_DISABLE_EC_SOFTWARE_SYNC
+	select GBB_FLAG_DISABLE_FWMP
+	select GBB_FLAG_DISABLE_LID_SHUTDOWN
+	select GBB_FLAG_DISABLE_PD_SOFTWARE_SYNC
+	select VBOOT_ALWAYS_ENABLE_DISPLAY
+	select VBOOT_NO_BOARD_SUPPORT
```
[the remark this answers is not in the document]
Comment (Stojek139808) [reply src/mainboard/protectli/vault_ehl/Kconfig:37]: Added `VBOOT_NO_TPM`, also added VBNV options that were in soc/ for elkhartlake, but are not there on this branch.

[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_ehl/Kconfig:37]: If the SOC doesn;t select those:
```
	select VBOOT_VBNV_CMOS
	select VBOOT_VBNV_CMOS_BACKUP_TO_FLASH
```

We don't have to select it either. We may simply select `VBOOT_VBNV_FLASH` for example and get rid of the CMOS which is troublesome.

[the remark this answers is not in the document]
Comment (Stojek139808) [reply src/mainboard/protectli/vault_ehl/Kconfig:37]: Fair enough, this builds too with just `VBOOT_VBNV_FLASH`, so pushed the change

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**